### PR TITLE
Improve and expand macro documentation.

### PR DESCRIPTION
**Pull request by jin**
Addressing [user feedback](https://twitter.com/AutomatedTester/status/1245695247801372672).

### CODE REVIEW

File: site/docs/skylark/macros.md [hunk 1]
```diff
@@ -7,12 +7,131 @@ title: Macros
...
+  )
+```
+
+You can also use macros to chain rules together. This example shows a generated
+text file undergoing conversions to markdown and rich text format:
+
+```python
+def txt_md_rtf_generator(name, visibility=None):
+  # Generates a txt file
+  native.genrule(
```
Comment (aiuto) [inline site/docs/skylark/macros.md:87]: Why bother with this. How about taking the text file name as an arg to the macro. That makes the example shorter without diminishing the point you are making.

File: site/docs/skylark/macros.md [hunk 2]
```diff
@@ -7,12 +7,131 @@ title: Macros
...
+  )
+  
+  # Converts the txt to markdown
+  native.genrule(
+    name = name + "-md",
+    srcs = [name + ".txt"],
+    outs = [name + ".md"],
+    cmd = "$(location //test:txt-to-md) $< > $@",
+    tools = ["//test:txt-to-md"],
+    visibility = visibility,
```
Comment (aiuto) [inline site/docs/skylark/macros.md:102]: You could illustrate another point here. What if the macro was  txt_to_rdf, and did not mention md.
You could give private visibility here to emphasize that the macro author only want the end result visible.

File: site/docs/skylark/macros.md [hunk 3]
```diff
@@ -7,12 +7,131 @@ title: Macros
...
+file_generator(
+    name = "file-three",
+    arg = "some_arg_three",
+)
+```
+
+Here, we are loading the `file_generator` symbol from a `.bzl` file located in the 
+`//path` package. By putting macro function definitions in a separate `.bzl` file, 
+we can keep our BUILD files clean and declarative,  The `.bzl` file can be loaded 
+from any package in the workspace, as long as `visibility` conditions are met.
```
Comment (laurentlb) [inline site/docs/skylark/macros.md:63]: No, visibility doesn't apply.

File: site/docs/skylark/macros.md [hunk 4]
```diff
@@ -7,12 +7,131 @@ title: Macros
...
+
+> Tip: To make a `.bzl` file visible from a package, add [`exports_files(["<filename>.bzl"])`](../be/functions.html#exports_files) into the `BUILD` file in the same directory as the `.bzl` file.
```
Comment (laurentlb) [inline site/docs/skylark/macros.md:65]: exports_files is not recommended

Comment (aiuto) [reply site/docs/skylark/macros.md:65]: Are you recommending dropping the paragraph or replacing it with a pointer to bzl_library?

Comment (jin) [reply site/docs/skylark/macros.md:65]: Not recommended at all? Also, it seems that we don't need it -- just an empty BUILD file will do.

Comment (jin) [reply site/docs/skylark/macros.md]: Thanks. Removed.

File: site/docs/skylark/macros.md (hunk 3 above)
Comment (jin) [reply site/docs/skylark/macros.md]: Removed.

File: site/docs/skylark/macros.md (hunk 1 above)
Comment (jin) [reply site/docs/skylark/macros.md]: The name is the arg to the macro. I simplified the example.

File: site/docs/skylark/macros.md (hunk 2 above)
Comment (jin) [reply site/docs/skylark/macros.md]: Done.

**Pull request closed without merging**